### PR TITLE
 #160452434 User reset password

### PR DESCRIPTION
**Pull request by hnobi**
#### What does this PR do?
This PR Adds reset password functionality to the codebase.
#### Description of Task to be completed?
- Add password reset form
- Adds passwordRestReducer reducer
- Adds actionType for resetPassword
- Adds resetPassword actions
- Adds resetPassword SCSS style

#### How should this be manually tested?

- Fetch and checkout to the branch
```$xslt
> git fetch
> git checkout  ft-160452434-reset-password-functionality
> npm install 
```
```$xslt
>  Rename the `.env.sample` file on the root directory of the cloned repo to `.env`. 

> Uncomment REACT_APP_HOST which is the link to the frontend application.

> REACT_APP_BASE_API_URL which is the base URL for the backend api. 

For example, `https://odin-ah-backend-staging.herokuapp.com/api/v1`.

> start the dev server with: `npm start`

>  run localhost/**/ ` in your browser and click on the `submit ` and check your email for reset password link.
```
#### Any background context you want to provide?
Nil 
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/160452434

#### Screenshots (if appropriate)
reset password

#### Questions:
Nil

### CODE REVIEW

**Comment by waleCloud** [comment]
Hello @hnobi, Well done on raising a PR. few things on your PR message;

- You need not mention  **Adds resetPasswordPage component** & **Adds resetPasswordform component**, resetPasswordPageComponent must definitely include the password form. 😄.

- On **How should this be manually tested**, you can endeavor to break the instructions into new lines rather having whoever is going to review scroll right/left, you can reduce the amount of details by focusing on just what's relevant for running the feature. For example

     - `cp .env.example .env`
     - Change the value of `REACT_APP_HOST` to `localhost/xx/xx`
     - Change the value of `REACT_APP_BASE_API_URL` to `https://odin-ah-backend-staging.herokuapp.com/api/v1`

**Review by waleCloud — COMMENTED** [review]
Increase test coverage, endeavor to cover all necessary files created/modified. before raising a final PR.
well done so far! 👍

**Comment by hnobi** [comment]
Thanks for the feedback @waleCloud. I will remove **Adds resetPasswordPage component & Adds resetPasswordform component** from my PR message now and I will increase the test coverage by covering all necessary files.

File: src/redux/actions/resetPassword.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            loading: false
+          }
+        });
+      });
+  }
+};
+
+
+export const completeResetRequest = data => (dispatch) => {
+  // const input = {password: data.password}
```
Comment (blackshady) [inline src/redux/actions/resetPassword.js]: Good job. Please remove the commented lines of code if you no longer need them.

Comment (hnobi) [reply src/redux/actions/resetPassword.js]: Thanks. I will remove the comment now.

**Review by Alameen688 — COMMENTED** [review]
Nice work 🎉 @hnobi. The reset password feature works as expected. 
However, it will be nice to have a success message that shows the new password has been set successfully, just before redirecting the user to login on the homepage.

**Pull request merged**